src/gui/torrentcontentmodel.cpp

```diff
@@ -469,16 +469,19 @@ void TorrentContentModel::setupModelData(const BitTorrent::TorrentInfo &info)
     // Iterate over files
     for (int i = 0; i < filesCount; ++i) {
         currentParent = m_rootItem;
-        QString path = Utils::Fs::toUniformPath(info.filePath(i));
+        const QString path = Utils::Fs::toUniformPath(info.filePath(i));
+
         // Iterate of parts of the path to create necessary folders
-        QStringList pathFolders = path.split('/', QString::SkipEmptyParts);
+        QVector<QStringRef> pathFolders = path.splitRef('/', QString::SkipEmptyParts);
         pathFolders.removeLast();
-        for (const QString &pathPart : asConst(pathFolders)) {
-            if (pathPart == ".unwanted")
+
+        for (const QStringRef &pathPart : asConst(pathFolders)) {
+            if (pathPart == QLatin1String(".unwanted"))
                 continue;
-            TorrentContentModelFolder* newParent = currentParent->childFolderWithName(pathPart);
+
+            TorrentContentModelFolder *newParent = currentParent->childFolderWithName(pathPart.toString());
             if (!newParent) {
-                newParent = new TorrentContentModelFolder(pathPart, currentParent);
+                newParent = new TorrentContentModelFolder(pathPart.toString(), currentParent);
```

Comment:
You kill all optimization when you create two strings from one string reference. Just store it when you create it first time.

Reply:
Fixed.